workflow-ui/core-android/src/main/java/com/squareup/workflow1/ui/container/ScreenOverlayDialogFactory.kt

```diff
@@ -59,7 +61,17 @@ public open class ScreenOverlayDialogFactory<S : Screen, O : ScreenOverlay<S>>(
     context: Context
   ): OverlayDialogHolder<O> {
     val contentViewHolder = initialRendering.content.toViewFactory(initialEnvironment)
-      .startShowing(initialRendering.content, initialEnvironment, context)
+      .startShowing(initialRendering.content, initialEnvironment, context) { view, doStart ->
+        // Note that we never call destroyOnDetach for this owner. That's okay because
+        // DialogSession.showNewDialog puts one in place above us on the decor view,
+        // and cleans it up. It's in place by the time we attach to the window, and
+        // so becomes our parent.
+        WorkflowLifecycleOwner.installOn(
+          view,
+          initialEnvironment.onBackPressedDispatcherOwner(view)
```

Comment:
This will pull in the wrong OnBackPressedDispatcherOwner, but that was always the case. This problem is why we've deprecated `ScreenOverlayDialogFactory` and replaced it with `fun ComponentDialog.setContent()` -- it was fundamentally broken, requiring us to create the content view before the `Dialog` that would host it.